sunpy/map/sources/mlso.py

```diff
@@ -35,10 +35,9 @@ def __init__(self, data, header, **kwargs):
         # Fill in some missing info
         self.meta['observatory'] = 'MLSO'
         self.meta['detector'] = 'KCor'
-        self.meta['waveunit'] = 'nanometer'
-        # Since KCor is on Earth, no need to raise the warning in mapbase
-        self.meta['dsun_obs'] = (sun.earth_distance(self.date)).to(u.m).value
-        self.meta['hgln_obs'] = 0.0
+        self._fix_and_warn('waveunit', 'nanometer')
+        self._fix_and_warn('dsun_obs', (sun.earth_distance(self.date)).to(u.m).value)
+        self._fix_and_warn('hgln_obs', 0)
```

Comment:
this makes me wonder, do we need to warn here? We know that this telescope is on Earth, and we know that the headers are missing this information (there is no standard enforcing they need to be there it's just part of sunpy's datamodel). Why are we going to throw warnings for this everytime?